### ISSUE TITLE
Mismatched declaration and definition of array type

### ISSUE THREAD

**Issue by thomthom** [PixarAnimationStudios/OpenSubdiv#1287]
I was testing Clang LLVM 15.0.5 on macOS. Caught this warning:

```
[build] /Users/thomas/SourceTree/subd/ThirdParty/OpenSubdiv/opensubdiv/vtr/level.cpp:1133:64: warning: argument 'points' of type 'OpenSubdiv::v3_5_0::Vtr::Index[12]' (aka 'int[12]') with mismatched bound [-Warray-parameter]
[build] Level::gatherTriRegularInteriorPatchPoints(Index fIndex, Index points[12], int rotation) const
[build]                                                                ^
[build] /Users/thomas/SourceTree/subd/ThirdParty/OpenSubdiv/opensubdiv/vtr/../vtr/level.h:371:71: note: previously declared as 'OpenSubdiv::v3_5_0::Vtr::Index[]' (aka 'int[]') here
[build]     int gatherTriRegularInteriorPatchPoints(      Index fIndex, Index patchVerts[], int rotation = 0) const;
[build]                                                                       ^
[build] 1 warning generated.
```

It's declared as:
```
int gatherTriRegularInteriorPatchPoints(      Index fIndex, Index patchVerts[], int rotation = 0) const;
```

But defined as:

```
int
Level::gatherTriRegularInteriorPatchPoints(Index fIndex, Index points[12], int rotation) const
```

Implementation suggests that the array is indeed expected to have exactly 12 items.

**Comment by davidgyu** [comment]
Filed as internal issue #OSD-399

**Comment by davidgyu** [comment]
Fixed by merging #1290
